### PR TITLE
apply eo-hamcrest to char, memory and random tests

### PR DESCRIPTION
**Pull request by Graur**
#660

### CODE REVIEW

File: eo-runtime/src/test/eo/org/eolang/char-tests.eo [hunk 1]
```diff
@@ -23,55 +23,63 @@
...
+  assert-that > @
     'A'.as-string
+    $.equal-to "A"
 
 [] > char-to-char-compares
-  eq. > @
-    'Ф'
-    'Ф'
+  assert-that > @
+    eq.
```
Comment (yegor256) [inline eo-runtime/src/test/eo/org/eolang/char-tests.eo:51]: @Graur everywhere in this file you use `equal-to`. Why do you use `eq` here?

File: eo-runtime/src/test/eo/org/eolang/char-tests.eo [hunk 2]
```diff
@@ -23,55 +23,63 @@
...
     eq.
       'x'
-      42
+      'x'
+    $.equal-to TRUE
+
+[] > compares-two-different-types
+  assert-that > @
+    not.
+      eq.
```
Comment (yegor256) [inline eo-runtime/src/test/eo/org/eolang/char-tests.eo:39]: @Graur same here, why not `equal-to`?

File: eo-runtime/src/test/eo/org/eolang/memory-tests.eo [hunk 3]
```diff
@@ -23,27 +23,34 @@
...
 +alias org.eolang.txt.sprintf
++alias org.eolang.hamcrest.assert-that
 +junit
 
 [] > writes-into-memory
   memory > x
-  and. > @
-    x.write 10
-    x.eq 10
+  assert-that > @
```
Comment (yegor256) [inline eo-runtime/src/test/eo/org/eolang/memory-tests.eo:31]: @Graur I think it should be done this way, to be fully in Hamcrest style:

```
[] > writes-into-memory
  assert-that > @
    seq.
      x.write 10
      x
    $.equal-to 10
```

File: eo-runtime/src/test/eo/org/eolang/memory-tests.eo [hunk 4]
```diff
@@ -23,27 +23,34 @@
...
-  and. > @
-    m.write "Hello, world!"
-    m.write 42
-    m.eq 42
+  assert-that > @
+    and.
+      m.write "Hello, world!"
+      m.write 42
+      m.eq 42
+    $.equal-to TRUE
```
Comment (yegor256) [inline eo-runtime/src/test/eo/org/eolang/memory-tests.eo:44]: @Graur the same here. `equal-to TRUE` is basically a cheating, we won't get any meaningful error message if `m` is not equal to 42.

**Comment by yegor256** [comment]
@Graur see above

**Comment by Graur** [comment]
@yegor256 I fixed some code according your comments. Please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**